crates/bevy_app/src/plugin.rs

```diff
@@ -144,12 +144,26 @@ mod sealed {
     impl<P: Plugin> Plugins<PluginMarker> for P {
         #[track_caller]
         fn add_to_app(self, app: &mut App) {
+            #[expect(
+                clippy::allow_attributes,
+                reason = "This lint only triggers sometimes, based on the features enabled."
+            )]
+            #[allow(unused_variables, reason = "plugin_name is only used with std enabled")]
```

Comment:
Rather than all of this "pomp and circumstance", can't we just name the bound field `_plugin_name`?

Reply:
or add a `let _ = plugin_name` to the end of the scope :)

Reply:
Or even better, learned from @janis-bhm: `#[allow(clippy::allow_attributes, unused_variables, reason = "...")]`